nemo/collections/multimodal/speech_llm/models/modular_models.py

```diff
@@ -836,7 +758,7 @@ def restore_from_pretrained_models(
             raise RuntimeError("PEFT training needs a trained base model present.")
 
         base_model_cfg = MegatronGPTSFTModel.merge_cfg_with(cfg.model.restore_from_path, cfg)
-        audio_model, audio_model_cfg = cls.get_audio_encoder_models_and_configs(cfg)
+        audio_model, audio_model_cfg = cls.get_audio_encoder_models_and_configs(cfg.model)
```

Comment:
try to be able to call get_audio_encoder_models_and_configs from setup_perception_modules

Reply:
it's a bit confusing to change the signature of `get_audio_encoder_models_and_configs`, since the `cfg` in other class methods like `load_adapters_for_inference`, `load_pretrained_audio_weights` and `load_audio_encoder_for_inference` refers to the input yaml config, not just the model config. It's better to change the `cfg` param of get_audio_encoder_models_and_configs to another name so as to avoid confusion

Reply:
reverted the change